src/imitation/algorithms/preference_comparisons.py

```diff
@@ -905,8 +927,30 @@ def __init__(
         self.optim = th.optim.AdamW(
             self._model.parameters(),
             lr=lr,
-            weight_decay=weight_decay,
         )
+        self.seed = seed
+        self.val_split = val_split
+
+        # TODO(juan) this could accept an arbitrary regularizer function
+        #  in the future if we wanted, with some changes in the __init__ arguments.
+        self.regularizer: Optional[regularization.Regularizer] = (
+            regularization.WeightDecayRegularizer(
+                initial_lambda=weight_decay,
+                optimizer=self.optim,
+                update_params_fn=weight_decay_updater
+                or regularization.ConstantParamScaler(),
+                logger=self.logger,
+            )
+            if weight_decay > 0
```

Comment:
Very long if-else statement makes it hard to understand flow. Could just set regularizer to None by default and then override it `if weight_decay > 0`.